specs/DOMContentLoaded.md

```diff
@@ -0,0 +1,119 @@
+# Background
+
+In response to consumers' requests for an event similar to the old [WebView DOMContentLoaded](https://docs.microsoft.com/en-us/microsoft-edge/hosting/webview#mswebviewdomcontentloaded), the WebView2 team has introduced DOMContentLoaded API which indicates that the main DOM elements have finished loading.
+In this document we describe the new API. We'd appreciate your feedback.
+
+# Description
+We propose adding DOMContentLoaded to CoreWebView2. This allows the developer to have an event that fires when the DOM is loaded after the WebView2 navigates to a page.
+
+# Examples
+## Win32 C++
+```
+ScenarioDOMContentLoaded::ScenarioDOMContentLoaded(AppWindow* appWindow)
+    : m_appWindow(appWindow), m_webView(appWindow->GetWebView())
+{
+    //! [DOMContentLoaded]
+    // Register a handler for the DOMContentLoaded event.
+    // Event is raised when the DOM content is loaded
+    CHECK_FAILURE(m_webView->add_DOMContentLoaded(
+        Callback<ICoreWebView2DOMContentLoadedEventHandler>(
+            [this](ICoreWebView2* sender, ICoreWebView2DOMContentLoadedEventArgs* args)
+                -> HRESULT {
+                m_webView->ExecuteScript(
+                    LR"~(
+                    let content=document.createElement("h2");
+                    content.style.color='blue';
+                    content.textContent="This text was added by the host app";
+                    document.body.appendChild(content);
+                    )~",
+                    Callback<ICoreWebView2ExecuteScriptCompletedHandler>(
+                        [](HRESULT error, PCWSTR result) -> HRESULT { return S_OK; })
+                        .Get());
+                return S_OK;
+            })
+            .Get(),
+        &m_DOMContentLoadedToken));
+    //! [DOMContentLoaded]
+```
+
+## C#
+```
+webView.CoreWebView2.DOMContentLoaded += (object sender, CoreWebView2DOMContentLoadedEventArgs arg) =>
+{
+    webView.ExecuteScriptAsync("let " +
```

Comment:
Does this need awaiting?

Reply:
If not, use a throwaway

 `_ = webView.ExeccuteScriptAsync...`.

Reply:
Awaiting is not necessary, use throwaway like Mike suggests.